packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -88,6 +89,67 @@ export const onSingleSelectHandler = <T extends Record<string, unknown>>(
   return newState;
 };
 
+/**
+ * Handles selection when clicked on a row while shift key is pressed.
+ */
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance?: TableInstance<T>,
+  onSelect?: (
+    selectedData: T[] | undefined,
+    tableState?: TableState<T>,
+  ) => void,
+  isRowDisabled?: (rowData: T) => boolean,
+) => {
+  if (instance == null) {
+    return state;
+  }
+
+  let startIndex = instance.flatRows.findIndex(
+    (row) => row.id === state.lastSelectedRow,
+  );
+  let endIndex = instance.flatRows.findIndex((row) => row.id === action.id);
+
+  if (startIndex < 0) {
+    startIndex = 0;
+  }
+  if (endIndex < 0) {
+    endIndex = 0;
+  }
+
+  if (startIndex > endIndex) {
+    const temp = startIndex;
+    startIndex = endIndex;
+    endIndex = temp;
+  }
+
+  const selectedRowIds: Record<string, boolean> = {};
+
+  // 1. Select all rows between start and end
+  instance.flatRows
+    .slice(startIndex, endIndex + 1)
+    .forEach((r) => (selectedRowIds[r.id] = true));
+
+  // 2. Select all children of the last row (endIndex)
+  const handleRow = (row: Row<T>) => {
+    selectedRowIds[row.id] = true;
+    row.subRows.forEach((r) => handleRow(r));
+  };
+  handleRow(instance.flatRows[endIndex]);
+
+  const newState = {
+    ...state,
+    selectedRowIds: selectedRowIds,
+  };
+
+  // 3.1 Deselect all selected disabled rows and their children
+  // 3.2 Convert all partially selected rows marked with tick mark to horizontal line
+  onSelectHandler(newState, instance, onSelect, isRowDisabled);
```

Comment:
What are these comments referring to? Because you are just calling the function here.

Reply:
I was just mentioning the reason for calling this function. Because I thought it may not be clear. I just noticed that there is a similar comment under [onSingleSelectHandler's line 86](https://github.com/iTwin/iTwinUI-react/blob/rohan/table-shift-click/packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts#L86).

But I can remove lines 146 and 147 if those comments are not necessary